src/sagemaker_containers/training.py

```diff
@@ -0,0 +1,138 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+from __future__ import absolute_import
+
+import importlib
+import os
+import shlex
+import subprocess
+
+import sagemaker_containers.training_environment as env
+
+_PYTHONPATH = 'PYTHONPATH'
+
+
+class Training(object):
```

Comment:
Two questions:

1). Can we design this so we don't lock users into this "framework"? I want to avoid the same problem we had with our original TensorFlow framework design. 

What would it look like if framework authors were implementing their train entrypoint using library functions that we provide? How much complexity would remain that we would still want to abstract away, at the cost of flexibility? I'd like to see us design that "API" first, and see some example user code in that style.

We may still want to offer something like this, which enforces a specific workflow for the entirety of training, which lets us ensure that all of our frameworks behave the same way. Even in that case, I'd argue that that should be written purely by calling those library functions.

2). For the "framework", is a class like this the best implementation? Other potential alternatives:
- A function which takes in a function/module
- A class which can be subclassed

Are there reasons for going this direction over the others?